pyiron_gui/project/project_browser.py

```diff
@@ -140,35 +272,68 @@ def __getattr__(self, item):
     def clear_output(self, *args, **kwargs):
         clear_button = kwargs.pop('clear_button', False)
         if clear_button:
-            self.buttons = widgets.HBox()
+            self.header = widgets.HBox()
         self.output.clear_output(*args, **kwargs)
         self.refresh()
 
-    def _update_buttons(self, obj):
+    def _click_button(self, b):
+        self.output.clear_output()
+        if b.description == "Show data":
+            b.description = "Show plot"
+            b.tooltip = "Display the plotting representation"
+            self.header.children = tuple([b])
+            self.refresh()
+            with self.output:
+                display(b.obj)
+        elif b.description == "Re-plot":
+            with self.output:
+                plt.ioff()
+                display(self._plot_array(b.obj))
+        else:
+            self.display(obj=b.obj)
 
-        def click_button(b):
-            self.output.clear_output()
-            self.display(obj=obj)
+    def _update_header(self, obj):
+        childs = []
 
-        if isinstance(obj, PyironWrapper) and obj.has_self_representation:
+        button = None
+        if isinstance(obj, BaseWrapper) and hasattr(obj, 'option_representation'):
+            childs.append(obj.option_representation)
+            button = widgets.Button(description="Apply")
+        elif isinstance(obj, BaseWrapper) and obj.has_self_representation:
             button = widgets.Button(description="Re-plot " + obj.name)
-            button.on_click(click_button)
-            self.buttons.children = tuple([button])
+        elif isinstance(obj, np.ndarray):
+            childs.append(self._array_plot_options(obj))
+            button = widgets.Button(description="Re-plot")
+            button.obj = obj
+            button.on_click(self._click_button)
```

Comment:
Ok, so these guys do have an `on_click` already! I guess I really need to learn a bit more about the widgets library before being able to give super useful advice. I remain concerned about the logic control being so dependent on the `description` string though.

Reply:
the `button.on_click` expects a method and passes the button itself to it. Therefore, I will refactor the `_click_button` method into several ones which do not use any if statements (and thus string comparisons) any more.

Reply:
Ok, I think I understand a bit better. I'm envisioning something like this now:

```python
button = widgets.Button(description="Re-plot")
button.obj = obj
button.on_click(self._click_replot_button)
```

Yeah? I am quite content with something of that flavour.

Reply:
Yes, that is the way I will do it now :) 
I just hope that something like 
```
def _click_show_data_button(self, button):
    button.description = 'Re-plot'
    button.on_click(self._click_replot_button)
```
will work 😅